### PR TITLE
MB-9097: Remove DPS 

### PR DESCRIPTION
**Pull request by ahobson**
## Summary

Remove dead code related to DPS.

Really this was grepping for DPS and SDDC and chasing the string. Lots of files changed, but not terribly complicated

## Setup to Run Your Code

```
make server_test
```
## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

### CODE REVIEW

**Comment by sheenamt** [comment]
@ahobson The things I can review (docker and env files) look good.

File: config/tls/devlocal-not-dps.key [hunk 1]
```diff
@@ -1,28 +0,0 @@
------BEGIN PRIVATE KEY-----
```
Comment (reggieriser) [inline config/tls/devlocal-not-dps.key:1]: **nitpick:** There's a reference to this file still in `.pre-commit-config.yaml`.

**Review by reggieriser — APPROVED** [review]
I started to review on behalf of DB reviewers, but went ahead and reviewed the whole PR (well, as much as makes sense for a big code pruning like this).  Things LGTM!  I did do a search for "dps" on the branch and found a few nitpicks that I added.  There's also a couple of references to `DPS users` in the top-level `README.md` which wasn't touched, so you may want to get rid of them while pruning.  But those aren't big deals, so going ahead and approving.  Thanks for taking care of this! 🚀

File: pkg/handlers/apitests.go [hunk 2]
```diff
@@ -169,19 +169,7 @@ func (suite *BaseHandlerTestSuite) AuthenticateOfficeRequest(req *http.Request,
...
-		ApplicationName: auth.MilApp,
-		UserID:          serviceMember.UserID,
-		IDToken:         "fake token",
-		DpsUserID:       dpsUser.ID,
-	}
-	ctx := auth.SetSessionInRequestContext(req, &session)
-	return req.WithContext(ctx)
-}
-
-// AuthenticateAdminRequest authenticates DPS users
```
Comment (gidjin) [inline pkg/handlers/apitests.go:184]: Praise: Nice catch on the copy pasta comment

File: pkg/models/client_cert_test.go [hunk 3]
```diff
@@ -21,13 +21,11 @@ func (suite *ModelSuite) Test_FetchClientCert() {
 	digest := "abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789"
 	subject := "/C=US/ST=DC/L=Washington/O=Test/OU=Test Cert/CN=localhost"
 	certNew := models.ClientCert{
-		Sha256Digest:    digest,
-		Subject:         subject,
-		AllowDpsAuthAPI: true,
-		AllowOrdersAPI:  true,
```
Comment (gidjin) [inline pkg/models/client_cert_test.go:27]: Question: Did you intend to delete the `AllowOrdersAPI` attribute here?

**Review by gidjin — CHANGES_REQUESTED** [review]
This looks good to me, I left one question for a line that looks like it shouldn't have been deleted so flagging this as requested changes since I'd like to make sure that line is correctly removed.

Second is a general question on this. Is there a plan/story to remove the environment vars from chamber after this PR is merged?

File: pkg/models/client_cert_test.go (hunk 3 above)
Comment (ahobson) [reply pkg/models/client_cert_test.go:27]: The test doesn't depend on the setting of any particular capability, so I thought I'd only add the minimum necessary to the model. Concerns with that approach?

**Comment by ahobson** [comment]
> This looks good to me, I left one question for a line that looks like it shouldn't have been deleted so flagging this as requested changes since I'd like to make sure that line is correctly removed.

I responded to the question as I think the line removed in the test was not necessary for the test and was arguably confusing.

> Second is a general question on this. Is there a plan/story to remove the environment vars from chamber after this PR is merged?

We have a ticket with infrasec related changes, but thank you for flagging that! I'll ask

File: pkg/models/client_cert_test.go (hunk 3 above)
Comment (gidjin) [reply pkg/models/client_cert_test.go:27]: I missed that this was a test! Thank for answering

**Review by gidjin — APPROVED** [review]

**Pull request merged**